src/tracking/particles.cpp

```diff
@@ -68,13 +69,16 @@ namespace impactx
 
         }
 
-        amrex::ParmParse const pp_algo("algo");
-        bool space_charge = false;
-        pp_algo.query("space_charge", space_charge);
+        auto space_charge = get_space_charge_algo();
```

Comment:
Double check if we skip the space charge on zero beam charge.

Reply:
Also: raise a warning if a user explicitly turns on space charge (or CSR) and then sets the charge to zero.

Reply:
Ah perfect, we have this warning already:
```
**** WARNINGS ******************************************************************
* GLOBAL warning list  after  [ FIRST STEP ]
*
* --> [!  ] [ElectrostaticSolver] [raised once]
*     Max norm of rho is 0
*     @ Raised by: ALL
*
* --> [!  ] [ImpactX::add_particles] [raised once]
*     The bunch charge is set to zero. ImpactX will run with zero-weighted
*     particles. Did you mean to set the space charge algorithm to off instead?
*     @ Raised by: ALL
*
********************************************************************************
```